### PR TITLE
Update Windows' build scripts for Visual Studio 2019

### PR DESCRIPTION
**Pull request by ghost**
**Description of work.**
This will tell the build servers to generate targets for, and build with, Visual Studio 16 2019. 

**To test:**
- Check the Windows server builds are successful.

*There is no associated issue.*

*This does not require release notes* because **internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/Jenkins/buildscript.bat [hunk 1]
```diff
@@ -185,7 +185,7 @@ if not "%JOB_NAME%"=="%JOB_NAME:debug=%" (
   set VATES_OPT_VAL=ON
 )
 
-call cmake.exe -G "%CM_GENERATOR%" -DCMAKE_SYSTEM_VERSION=%SDK_VERS% -DCONSOLE=OFF -DENABLE_CPACK=ON -DMAKE_VATES=%VATES_OPT_VAL% -DParaView_DIR=%PARAVIEW_DIR% -DMANTID_DATA_STORE=!MANTID_DATA_STORE! -DUSE_PRECOMPILED_HEADERS=ON %PACKAGE_OPTS% ..
+call cmake.exe -G "%CM_GENERATOR%" -A %CM_ARCH% -DCMAKE_SYSTEM_VERSION=%SDK_VERS% -DCONSOLE=OFF -DENABLE_CPACK=ON -DMAKE_VATES=%VATES_OPT_VAL% -DParaView_DIR=%PARAVIEW_DIR% -DMANTID_DATA_STORE=!MANTID_DATA_STORE! -DUSE_PRECOMPILED_HEADERS=ON %PACKAGE_OPTS% ..
```
Comment (martyngigg) [inline buildconfig/Jenkins/buildscript.bat]: https://social.msdn.microsoft.com/Forums/en-US/934aa55e-1641-4858-a7a5-a0c9bed4f85d/project-file-vctargetspatchvcxproj-cannot-find-visual-studio-2019-build-tools-even-though-i-have?forum=msbuild suggests that the `-A` needs to be quoted as: `"-A %CM_ARCH%"` or the solution won't build.

Comment (hsautessella) [reply buildconfig/Jenkins/buildscript.bat]: `Win64` isn't listed in the [CMake 13.5 docs](https://cmake.org/cmake/help/git-stage/generator/Visual%20Studio%2016%202019.html) as an architecture; I've changed it to `x64`

Comment (martyngigg) [reply buildconfig/Jenkins/buildscript.bat]: Good call. :+1:

File: buildconfig/Jenkins/buildscript.bat [hunk 2]
```diff
@@ -185,7 +185,7 @@ if not "%JOB_NAME%"=="%JOB_NAME:debug=%" (
   set VATES_OPT_VAL=ON
 )
 
-call cmake.exe -G "%CM_GENERATOR%" -DCMAKE_SYSTEM_VERSION=%SDK_VERS% -DCONSOLE=OFF -DENABLE_CPACK=ON -DMAKE_VATES=%VATES_OPT_VAL% -DParaView_DIR=%PARAVIEW_DIR% -DMANTID_DATA_STORE=!MANTID_DATA_STORE! -DUSE_PRECOMPILED_HEADERS=ON %PACKAGE_OPTS% ..
+call cmake.exe -G "%CM_GENERATOR%" "-A %CM_ARCH%" -DCMAKE_SYSTEM_VERSION=%SDK_VERS% -DCONSOLE=OFF -DENABLE_CPACK=ON -DMAKE_VATES=%VATES_OPT_VAL% -DParaView_DIR=%PARAVIEW_DIR% -DMANTID_DATA_STORE=!MANTID_DATA_STORE! -DUSE_PRECOMPILED_HEADERS=ON %PACKAGE_OPTS% ..
```
Comment (martyngigg) [inline buildconfig/Jenkins/buildscript.bat]: Maybe the quotes aren't necessary after all? - https://cmake.org/cmake/help/v3.15/generator/Visual%20Studio%2016%202019.html#platform-selection

**Review by martyngigg — APPROVED** [review]
Builds are now passing. Looks good to go.

**Comment by martyngigg** [comment]
@mantidproject/gatekeepers Any chance we could get this merged for the clean builds tonight?

**Pull request merged**